### DISCUSSION TITLE
New tuya temperature & humidity device

### DISCUSSION THREAD

**Discussion by kasi0815** [Koenkk/zigbee2mqtt#7283] [Devices]
Hi, I bought this [device ](https://de.aliexpress.com/item/1005002281480470.html) and added it as it is not supported so far.

I added to devices.js:

`    {
    	fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_bq5c8xfe'}],
        model: '',   //empty as I do not know it
        vendor: 'Tuya',
        description: 'Temperature & humidity sensor',
        supports: 'humidity, temperature',
        fromZigbee: [
            fz.ignore_basic_report, // Add this if you are getting no converter for 'genBasic'
            fz.maa_tuya_temp_sensor, 
        ],
        toZigbee: [],
        exposes: [e.temperature(), e.humidity(), e.battery()],
    },
`

and to fromZigbee.js:

`    maa_tuya_temp_sensor: {
        cluster: 'manuSpecificTuya',
        type: ['commandSetDataResponse', 'commandGetData'],
        convert: (model, msg, publish, options, meta) => {
            const dp = msg.data.dp;  // First we get the data point ID
            const value = tuya.getDataValue(msg.data.datatype, msg.data.data); // This function will take of converting the data to proper JS type
            switch (dp) {
            case 1:
                return {temperature: (value / 10).toFixed(1)}; // value is already converted to a number in JS, and we deduced that it needs to be divided by 10
            case 2:
                return {humidity: value};
            case 3:
 //               return {linkquality: value};	//linkquality? temperature_precision?
                break;
            case 4:
                return {battery: value};
            default:
                meta.logger.warn(`zigbee-herdsman-converters:maa_tuya_temp_sensor: NOT RECOGNIZED DP #${dp} with data ${JSON.stringify(msg.data)}`); // This will cause zigbee2mqtt to print similar data to what is dumped in tuya.dump.txt
            }
        },
    },
`

The communication looks now like this:

`{"battery":100,"device":{"applicationVersion":70,"dateCode":"","friendlyName":"KlimaLukas","hardwareVersion":1,"ieeeAddr":"0x847127fffe0d0505","manufacturerID":4098,"manufacturerName":"_TZE200_bq5c8xfe","model":"???","networkAddress":1635,"powerSource":"Battery","stackVersion":0,"type":"EndDevice","zclVersion":3},"humidity":43,"linkquality":18,"temperature":"24.1"}`

and I can use those values in OpenHAB since some weeks. 

It is the first time that I integrated a not supported device. My code is not perfect for sure because I'm not a programmer. However I hope it helps somebody.

Have fun.   :-)

**Comment by Koenkk** [comment]
Added!

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

**Reply by kasi0815** [reply]
Great, thank you!   :-)

**Reply by kasi0815** [reply]
I was just curious how you implemented the part in the file fromZigbee.js. I think there is a unsightly part inside. The line 

`       default:
                meta.logger.warn(`zigbee-herdsman-converters:maa_tuya_temp_sensor: NOT RECOGNIZED ` +
                    `DP #${dp} with data ${JSON.stringify(msg.data)}`);
`

should be replaced by something like this:

`       default:
                meta.logger.warn(`zigbee-herdsman-converters:**tuya_temperature_humidity_sensor**: NOT RECOGNIZED ` +
                    `DP #${dp} with data ${JSON.stringify(msg.data)}`);
`
